monitoring/asset_proof_collector.go

```diff
@@ -0,0 +1,100 @@
+package monitoring
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	numAssetProofsMetric = "asset_proofs"
+)
+
+// assetProofCollector is a Prometheus collector that exports proof related data
+// of all taproot assets.
+type assetProofCollector struct {
+	collectMx sync.Mutex
+
+	cfg      *PrometheusConfig
+	registry *prometheus.Registry
+
+	proofsVec *prometheus.GaugeVec
+}
+
+func newAssetProofCollector(cfg *PrometheusConfig,
+	registry *prometheus.Registry) (*assetProofCollector, error) {
+
+	if cfg == nil {
+		return nil, errors.New("asset proof collector prometheus cfg " +
+			"is nil")
+	}
+
+	if cfg.AssetStore == nil {
+		return nil, errors.New("asset proof collector asset store is " +
+			"nil")
+	}
+
+	return &assetProofCollector{
+		cfg:      cfg,
+		registry: registry,
+		proofsVec: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Name: numAssetProofsMetric,
```

Comment:
This would be another good candidate for a histogram. We'd then be able to plot percentiles to find outliers.